### PR TITLE
Header cleanup, remove shebang and utf8 lines

### LINKED ISSUE #3041: Remove unnecessary shebang in Python files
Hi,

While preparing a simple pull request for #3040 , I gave a step back to look at `xrandom.py` and see what else that file had. And noticed the `#/usr/bin/env python3` shebang, present in most (all?) of Cylc files.

As far as I know, that shebang is useful only for the scripts that may be executed directly (e.g. `./myscript.py`). So in the case of `xrandom.py` it implies that that script could be executed with `python3`, but actually that file is not supposed to be used that way, but instead only via Python import module.

I think the shebangs could be removed, which would save some commits when Python 4 is released too. And maybe we should include `# -*- coding: utf-8 -*-`? I believe `isodatetime` does that already.

Thoughts?

### PR DESCRIPTION
**Pull request by sgaist**
These changes close #3041 

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Already covered by existing tests.
- [x] No change log entry required -> invisible to users
- [x] No documentation update required.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -22,7 +22,7 @@ Issue](https://github.com/cylc/cylc-flow/issues) as the team may not be able to
 consider large changes that appear out of the blue. New contributors should
 add their details to the [Code Contributors](#code-contributors) section of
 this file as part of their first Pull Request, and reviewers are responsible
-for checking this before merging the new branch into *master*. 
+for checking this before merging the new branch into *master*.
```
Comment (kinow) [inline CONTRIBUTING.md]: Had a look at the rendered markdown, and both the link above and this line are both correctly displayed. So I think removing the leading space should be OK.

File: .travis/cover.py [hunk 2]
```diff
@@ -1,5 +1,3 @@
-#!/usr/bin/env python3
-
```
Comment (kinow) [inline .travis/cover.py]: Oh, had completely forgotten about these Python files, good catch. Let's just wait and confirm Travis can still execute/use them correctly.

**Comment by kinow** [comment]
(Updated title to include that this one is about the shebang and utf8 lines)

File: setup.py [hunk 3]
```diff
@@ -1,6 +1,3 @@
-#!/usr/bin/env python
-# coding=utf-8
```
[the remark this answers is not in the document]
Comment (sgaist) [reply setup.py]: You're right, I thought I had undone the changes but they were stashed.

File: CONTRIBUTING.md (hunk 1 above)
Comment (sgaist) [reply CONTRIBUTING.md]: That wasn't intended, my editor is configured to clean withe spaces at line ends and I managed to miss that file going in.

In any case,  I don't mind changing back as it is not directly related to the cleanup at hand.

**Comment by kinow** [comment]
Hmmm, three out of four builds in Travis for the functional tests have failed already. Will take a better look tomorrow, but the other PR's tests seem to be going well so far.

File: tests/parsec/synonyms/bin/synonyms.py [hunk 4]
```diff
@@ -1,5 +1,3 @@
-#!/usr/bin/env python3
```
Comment (kinow) [inline tests/parsec/synonyms/bin/synonyms.py]: Looks removing the shebang in test files causes the functional tests to fail.

All 4 functional test builds failed in Travis @sgaist. Unfortunately they timed out, so there is no helpful output there (:angry:). So I've executed one that failed locally (`./tests/parsec/synonyms/05-float_list.t`) and it seems to have failed as the shell script will try to actually run the python script. See screenshot of the test:

image

Could you revert the change for the `./tests/` directory, please @sgaist ? Thanks!

Comment (sgaist) [reply tests/parsec/synonyms/bin/synonyms.py]: Sure thing

**Comment by kinow** [comment]
Had to kick one of our Travis build as it had one failed test. @sgaist this one looks ready to be double-reviewed/merged. Could you just fix the latest merging conflicts, please?

**Comment by kinow** [comment]
Thanks for updating it @sgaist . The important builds have passed already on Travis. I'm just waiting for the sensitive tests (which run flaky tests). They should finish any moment now. Then will wait a bit more for Codecov bot to report the coverage (as this change touched the files that are used to generate the coverage reports). But if everything goes well, it should be approved in the next minutes!

Thanks a ton for both of your contributions, and for your patience while working on these issues :+1: 
Bruno

**Comment by kinow** [comment]
@cylc/core just needs one more review, and should be ready to merge.

**Comment by hjoliver** [comment]
Thanks @sgaist :bouquet:

**Pull request merged**